src/IO/WriteBufferFromS3.cpp

```diff
@@ -288,12 +288,15 @@ WriteBufferFromS3::~WriteBufferFromS3()
 
     if (canceled)
     {
-        LOG_INFO(
-            log,
-            "WriteBufferFromS3 was canceled."
-            "The file might not be written to S3. "
-            "{}.",
-            getVerboseLogDetails());
+        if (!isEmpty())
+        {
+            LOG_INFO(
+                log,
+                "WriteBufferFromS3 was canceled."
+                "The file might not be written to S3. "
+                "{}.",
+                getVerboseLogDetails());
+        }
```

Comment:
The code for Azure write buffer somewhat mirrored: https://github.com/ClickHouse/ClickHouse/blob/master/src/Disks/IO/WriteBufferFromAzureBlobStorage.cpp#L89-L94

It makes sense to modify logging there as well.